.cbmc-batch/jobs/aws_array_list_swap_contents/aws_array_list_swap_contents_harness.c

```diff
@@ -0,0 +1,16 @@
+#include <aws/common/array_list.h>
+#include <proof_helpers.h>
+
+void aws_array_list_swap_contents_harness() {
+    struct aws_array_list *from;
+    ASSUME_BOUNDED_ARRAY_LIST(from, MAX_INITIAL_ITEM_ALLOCATION, MAX_ITEM_SIZE);
+
+    struct aws_array_list *to;
+    ASSUME_BOUNDED_ARRAY_LIST(to, MAX_INITIAL_ITEM_ALLOCATION, MAX_ITEM_SIZE);
+
+    __CPROVER_assume(from->alloc != NULL);
+    __CPROVER_assume(from->alloc == to->alloc);
```

Comment:
Is this necessary?

Reply:
Yes, because the source code has this [assertion](https://github.com/awslabs/aws-c-common/blob/f0968526f65d7e72ea2d4f66f2b02e1faeb38198/include/aws/common/array_list.inl#L184)